utils/node_local_test.py

```diff
@@ -3988,6 +3988,77 @@ def test_pydaos_kv(server, conf):
     daos._cleanup()
     log_test(conf, pydaos_log_file.name)
 
+
+def test_pydaos_kv_obj_class(server, conf):
+    """Test the predefined object class works with KV"""
+    # pylint: disable=consider-using-with
+
+    pydaos_log_file = tempfile.NamedTemporaryFile(prefix='kv_objclass_pydaos_',
+                                                  suffix='.log',
+                                                  delete=False)
+
+    os.environ['D_LOG_FILE'] = pydaos_log_file.name
+    daos = import_daos(server, conf)
+
+    pool = server.get_test_pool_obj()
+
+    cont = create_cont(conf, pool, ctype="PYTHON")
+
+    container = daos.DCont(pool.uuid, cont.uuid)
+    failed = False
+    # Write kv1 dictionary with OC_S2 object type
+    kv1 = container.dict('object1', {"Monday": "1"}, "OC_S2")
+    if len(kv1) != 1:
+        failed = True
+        print(f'Expected length of kv object is 1 but got {len(kv1)}')
+
+    # Write kv2 dictionary without any object type,
+    # so in this case we have 4 targets so default object type should be S4
+    kv2 = container.dict('object2', {"Monday": "1", "Tuesday": "2"})
+    if len(kv2) != 2:
+        failed = True
+        print(f'Expected length of kv object is 2 but got {len(kv2)}')
+
+    # Run a command to list the objects
+    cmd = ['cont', 'list-objects', pool.uuid, cont.uuid]
+    print('list the objects from container')
+    rc = run_daos_cmd(conf, cmd, use_json=True)
+
+    data = rc.json
+    assert data['status'] == 0, rc
+    assert data['error'] is None, rc
+    assert data['response'] is not None, rc
+
+    # Run a command to get the object layout
+    print('query the object layout')
+    actual_obj_layout = []
+    for obj in data['response']:
+        cmd = ['object', 'query', pool.uuid, cont.uuid, obj]
+        rc = run_daos_cmd(conf, cmd, use_json=True)
+
+        query_data = rc.json
+        assert query_data['status'] == 0, rc
+        assert query_data['error'] is None, rc
+        assert query_data['response'] is not None, rc
+        actual_obj_layout.append(query_data['response']['class'])
+
+    # Verify the object has the correct layout used during kv dictionary creation.
+    expected_obj_layout = ['S2', 'S4']
+    for obj in expected_obj_layout:
+        if obj not in actual_obj_layout:
+            failed = True
+            print(f'Expected obj {obj} not found in all {actual_obj_layout}')
+
+    if failed:
+        print("Test step failed in test_pydaos_kv_obj_class()")
```

Comment:
This will only print to stdout but not actually fail anything.  You need something like the below to fail NLT.

`conf.wf.add_test_case('pydaos object test', failure='test failed')`

Reply:
I will update

Reply:
Thinking about this more you want to create the test case anyway, rather than just on failure so you want something like:
`if failed:
   conf.wf.add_test_case('pydaos object test', failure='test failed')
else:
   conf.wf.add_test_case('pydaos object test')
`
and then you'll be able to tell from the test results that this code ran.

I note that the existing test_pydaos_kv() function does not do this, it long predates the junit reporting of results so it does warrant updating.  If you don't want to do it here that's fine, I can push a PR to bring that one up to spec once this PR is compete.

Reply:
I have not updated all the review comments, so I can incorporate this as well. It's a good way to confirm. 
I used the dummy prints just to make sure it ran in Ci. 
I am not sure how this NLT results are populating in Jenkins, so anything gives the test confirmation will on console will be idle.